crates/control/src/localization.rs

```diff
@@ -895,67 +912,71 @@ fn get_translation_and_rotation_measurement(
         _ => field_mark_correspondence.measured_line_in_field,
     };
     let measured_line_in_field_vector = measured_line_in_field.1 - measured_line_in_field.0;
-    let signed_distance_to_line = measured_line_in_field
-        .signed_distance_to_point(Point2::from(robot_to_field.translation.vector));
+    let signed_distance_to_line =
+        measured_line_in_field.signed_distance_to_point(ground_to_field * Point2::origin());
     match field_mark_line_direction {
         Direction::PositiveX => {
-            vector![
-                field_mark_line.0.y + signed_distance_to_line,
-                (-measured_line_in_field_vector.y).atan2(measured_line_in_field_vector.x)
-                    + robot_to_field.rotation.angle()
+            nalgebra::vector![
+                field_mark_line.0.y() + signed_distance_to_line,
+                (-measured_line_in_field_vector.y()).atan2(measured_line_in_field_vector.x())
```

Comment:
you said something about replacing atan?

Reply:
Yes, when passing the result of the atan2 directly to a UnitComplex constructor or something like that.
However, in this area of the code, atan2 seems to be used in non-standard ways, swapping the arguments or inverting axes.
I'm not convinced replacing atan2 here makes things better necessarily.